frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -314,6 +314,30 @@ export const getPipelineRunWorkspaces = (
   );
 };
 
+export const calculateRelativeTime = (startTime: string, completionTime?: string) => {
+  const start = new Date(startTime).getTime();
+  const end = completionTime ? new Date(completionTime).getTime() : new Date().getTime();
+  const secondsAgo = (end - start) / 1000;
+  const minutesAgo = secondsAgo / 60;
+  const hoursAgo = minutesAgo / 60;
+
+  if (minutesAgo > 90) {
+    const count = Math.round(hoursAgo);
+    return `about ${count} hours`;
+  }
+  if (minutesAgo > 45) {
+    return 'less than a hour';
```

Comment:
English is stupid lol. I read this and felt it was wrong, but I didn't want to say why without finding a reason for it. I think this qualities:

> First thing first, to decide whether it’s a/an to a word, it all depends on how we pronounce the word itself. If we pronounce the word, and the first sound starts with vowels (a,i,u,o,e) you give ‘an’. [...]
> 
> For the word ‘Hour’, the correct pronunciation is /auer/ more or less. The first letter we hear is ‘a’ which is a vowel, so we put ‘an hour’ not ‘a hour’. I am giving you more example. Let’s take a look at the word ‘University’, to pronounce it, you say /yuniversiti/. Having said that, the very first letter you her is /y/ so you do not put ‘an university’, but ‘a university’.

[[source](https://www.quora.com/Which-is-correct-a-hour-or-an-hour)]

```suggestion
    return 'less than an hour';
```